components/config/opts.rs

```diff
@@ -518,6 +521,7 @@ pub fn default_opts() -> Opts {
         userscripts: None,
         user_stylesheets: Vec::new(),
         output_file: None,
+        max_session_history: 256,
```

Comment:
Number rather chosen at random. I'm not sure how to make a rational choice for this number, especially since the choice is probably very dependent on hardware: the main resource I expect we're saving is swap space (since frozen threads don't have timers so shouldn't be swapping themselves back in) which is very different, e.g. on a phone vs a PC.